compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -330,7 +330,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         // Don't show generic arguments when the method can't be found in any implementation (#81576).
         let mut ty_str_reported = if trait_missing_method {
-            ty_str.strip_prefix("dyn ").expect("Failed to remove the prefix dyn").to_owned()
+            let ty::Dynamic(data, _, _) = rcvr_ty.kind() else {
+                bug!("Failed to get the trait name of the bare object")
+            };
+            data.to_string()
         } else {
             ty_str.clone()
         };
```

Comment:
Could this directly mutate `ty_str` (and eventually short_ty_str`), to avoid having this code a second time later, and avoid the `bug!`?